File: sdkwork-rtc-sdk-python/sdkwork_rtc_sdk/capability_catalog.py
from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class RtcCapabilityCatalogEntry:
    capabilityKey: str
    category: str
    surface: str


class RtcCapabilityCatalog:
    entries = [
        RtcCapabilityCatalogEntry("session", "required-baseline", "cross-surface"),
        RtcCapabilityCatalogEntry("join", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("publish", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("subscribe", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("mute", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("basic-events", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("health", "required-baseline", "control-plane"),
        RtcCapabilityCatalogEntry("unwrap", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("screen-share", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("recording", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("cloud-mix", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("cdn-relay", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("data-channel", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("transcription", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("beauty", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("spatial-audio", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("e2ee", "optional-advanced", "runtime-bridge"),
    ]


def get_rtc_capability_catalog() -> list[RtcCapabilityCatalogEntry]:
    return RtcCapabilityCatalog.entries


def get_rtc_capability_descriptor(capability_key: str) -> Optional[RtcCapabilityCatalogEntry]:
    for entry in RtcCapabilityCatalog.entries:
        if entry.capabilityKey == capability_key:
            return entry

    return None

File: sdkwork-rtc-sdk-go/rtcstandard/capability_catalog.go
package rtcstandard

type RtcCapabilityCatalogEntry struct {
    CapabilityKey string
    Category      string
    Surface       string
}

type RtcCapabilityCatalog struct{}

var RTC_CAPABILITY_CATALOG = []RtcCapabilityCatalogEntry{
    {CapabilityKey: "session", Category: "required-baseline", Surface: "cross-surface"},
    {CapabilityKey: "join", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "publish", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "subscribe", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "mute", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "basic-events", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "health", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "unwrap", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "screen-share", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "recording", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "cloud-mix", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "cdn-relay", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "data-channel", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "transcription", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "beauty", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "spatial-audio", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "e2ee", Category: "optional-advanced", Surface: "runtime-bridge"},
}

func GetRtcCapabilityCatalog() []RtcCapabilityCatalogEntry {
    return append([]RtcCapabilityCatalogEntry(nil), RTC_CAPABILITY_CATALOG...)
}

func GetRtcCapabilityDescriptor(capabilityKey string) *RtcCapabilityCatalogEntry {
    for index := range RTC_CAPABILITY_CATALOG {
        if RTC_CAPABILITY_CATALOG[index].CapabilityKey == capabilityKey {
            return &RTC_CAPABILITY_CATALOG[index]
        }
    }

    return nil
}
